### PR TITLE
Mask client secret and initiator password in debug

### PR DESCRIPTION
**Pull request by collinsmuriuki**
Prevent client secret and initiator password visibility in `Mpesa` client debug

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -524,6 +515,11 @@ impl<'mpesa, Env: ApiEnvironment> Mpesa<Env> {
     }
 }
 
+#[derive(Debug, serde::Deserialize)]
+struct AuthResponse {
```
Comment (itsyaasir) [inline src/client.rs:519]: You are missing an extra field here which is `expiry_in`, if you want to ignore this value, derive serde default as well

**Comment by itsyaasir** [comment]
Hey Collins, I think there's another approach to hide the secret key, we can derive manual `Debug` on the `MpesaClient` debug and remove the fields from there.

File: src/client.rs (hunk 1 above)
Comment (collinsmuriuki) [reply src/client.rs:519]: I had actually included it but I realized I was not using the value anyway, at least in this PR. You can include it in #78 if you will need to reference it there

Comment (collinsmuriuki) [reply src/client.rs:519]: WDYT?

Comment (collinsmuriuki) [reply src/client.rs:519]: I just checked out #78, I think I could omit my changes in favor of yours just to avoid any mix up

**Comment by collinsmuriuki** [comment]
> Hey Collins, I think there's another approach to hide the secret key, we can derive manual `Debug` on the `MpesaClient` debug and remove the fields from there.

Thought of this too, but the `secrecy` crate provides the ability to wipe the secrets from memory on `drop` securely, this was the main reason I chose to go with the crate

**Comment by itsyaasir** [comment]
> > Hey Collins, I think there's another approach to hide the secret key, we can derive manual `Debug` on the `MpesaClient` debug and remove the fields from there.
> 
> Thought of this too, but the `secrecy` crate provides the ability to wipe the secrets from memory on `drop` securely, this was the main reason I chose to go with the crate

That is great, We can go for this

**Review by itsyaasir — APPROVED** [review]

**Pull request merged**